pkg/util/webhooks/tls.go

```diff
@@ -169,22 +176,29 @@ func SetupTLSForVirtHandlerClients(caManager ClientCAManager, certManager certif
 				return fmt.Errorf("no client certificate provided.")
 			}
 
-			c, err := x509.ParseCertificate(rawCerts[0])
+			rawClient, rawIntermediates := rawCerts[0], rawCerts[1:]
+			c, err := x509.ParseCertificate(rawClient)
 			if err != nil {
 				return fmt.Errorf("failed to parse peer certificate: %v", err)
 			}
-			_, err = c.Verify(x509.VerifyOptions{
-				Roots:     certPool,
-				KeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
-			})
 
-			if err != nil {
-				return err
+			var intermediatePool *x509.CertPool = nil
+			if externallyManaged {
```

Comment:
Isn't that a duplicate to line 120? If so, extract this into a function, please.

Reply:
There is only a small portion of overlap. I tried to make the minimal change to the code.

Reply:
But you added code blocks that are duplicating each other. Maybe just extract this into a utility function and call it? I.e. like this: https://gist.github.com/dhiller/cb52a0c0e13e8107f544afa611793cb8

Reply:
Refactored.